### ISSUE TITLE
Why has the CLI worker command flag `optimizations` changed from `--optimizations` in 5.*?

### ISSUE THREAD

**Issue by stuart-bradley** [celery/celery#8224]
This question is referencing this code:

https://github.com/celery/celery/blob/3b1c768c39f641a70d3086bd5a1079a421f94344/celery/bin/worker.py#L169
```
@click.option('optimization',
              '-O',
              default='default',
              cls=CeleryOption,
              type=click.Choice(('default', 'fair')),
              help_group="Worker Options",
              help="Apply optimization profile.")
```
Why has the flag removed the double hyphen? It seems like a bug, but I assume there was a reason for it? The reason I'm asking is that this change breaks a bunch of code we have, so I'd like to understand the change.

**Comment by auvipy** [comment]
@thedrow was this an intended change? @stuart-bradley  we actually moved the CLI from argparse to click in version 5 on wards, so that might cause it. or it could be a typo

**Comment by stuart-bradley** [comment]
Hey team, has there been any movement on this?

**Comment by auvipy** [comment]
we switched from argparse to click library. may be that could be a reason

**Comment by stuart-bradley** [comment]
Made the fix myself if someone can review: #8312
